libsolidity/interface/StandardCompiler.cpp

```diff
@@ -850,15 +850,87 @@ Json::Value StandardCompiler::compileSolidity(StandardCompiler::InputsAndSetting
 	return output;
 }
 
+
+Json::Value StandardCompiler::compileYul(InputsAndSettings _inputsAndSettings)
+{
+	if (_inputsAndSettings.sources.size() != 1)
+		return formatFatalError("JSONError", "Yul mode only supports exactly one input file.");
+	if (!_inputsAndSettings.smtLib2Responses.empty())
+		return formatFatalError("JSONError", "Yul mode does not support smtlib2responses.");
+	if (!_inputsAndSettings.remappings.empty())
+		return formatFatalError("JSONError", "Field \"settings.remappings\" cannot be used for Yul.");
+	if (!_inputsAndSettings.libraries.empty())
+		return formatFatalError("JSONError", "Field \"settings.libraries\" cannot be used for Yul.");
+
```

Comment:
Should also reject `metadata`.

Reply:
Hm, what exactly do you mean? This would be part of "outputSelection" - which we do not really validate...

Reply:
There is a setting block for metadata. `useLiteralContent` is its only field right now.

Reply:
OK, but I cannot check it at this point anymore, because only a 'bool' is retained.

Reply:
Ok. If we want to be pedantic just raise an error if that bool is to true.